### ISSUE TITLE
Unable to filter on dimensions with lookups

### ISSUE THREAD

**Issue by hollow** [allegro/turnilo#224]
I'm trying to convert our pivot installation to turnilo and the basic setup works. However, when I'm trying to filter on a dimension that uses a lookup, the query is either built incorrectly or cannot be built at all and throws a `cannot resolve $dimension` error.

The lookup is pretty simple, e.g. we have an `organization` dimension in druid containing IDs that map to a human-readable name using the ID as a primary key into our PostgreSQL database (where Druid loads the lookups from). In Pivot this was working as expected:

```
- name: organization
  title: 'Organization'
  formula: $organization.lookup('organization_display_value').fallback($organization)
```

In Turnilo this returns an empty result when filtering or pinning an organization because the lookup value is used in the druid query filter instead of the original dimension value:

```
[
  [
    {
      "queryType": "timeseries",
      "dataSource": "bids",
      "intervals": "2018-11-05T07:02Z/2018-11-06T07:02Z",
      "granularity": "all",
      "filter": {
        "type": "selector",
        "dimension": "organization",
        "value": "[1] Test (DONT DELETE)"
      },
      "aggregations": [
        {
          "name": "__VALUE__",
          "type": "longSum",
          "fieldName": "clicks"
        }
      ]
    }
  ]
]
```

I've also tried to not clobber the dimension name, like so:

```
- name: organization_display_value
  title: 'Organization'
  formula: $organization.lookup('organization_display_value').fallback($organization)
```

However, this way the query cannot even be built by turnilo:

```
Nov 06 07:09:02 master1 turnilo[17632]: Error: could not resolve $organization_display_value
Nov 06 07:09:02 master1 turnilo[17632]:     at RefExpression.changeInTypeContext (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5402:19)
Nov 06 07:09:02 master1 turnilo[17632]:     at RefExpression.<anonymous> (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:4338:27)
Nov 06 07:09:02 master1 turnilo[17632]:     at RefExpression.Expression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:3883:34)
Nov 06 07:09:02 master1 turnilo[17632]:     at OverlapExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5008:40)
Nov 06 07:09:02 master1 turnilo[17632]:     at AndExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5010:46)
Nov 06 07:09:02 master1 turnilo[17632]:     at FilterExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5010:46)
Nov 06 07:09:02 master1 turnilo[17632]:     at ApplyExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5010:46)
Nov 06 07:09:02 master1 turnilo[17632]:     at ApplyExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5008:40)
Nov 06 07:09:02 master1 turnilo[17632]:     at ApplyExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5008:40)
Nov 06 07:09:02 master1 turnilo[17632]:     at ApplyExpression.ChainableUnaryExpression._substituteHelper (/var/app/turnilo/releases/eef4905a3247dc2439338e4e7a4fbc44b16a785d/node_modules/plywood/build/plywood.js:5008:40)
```

**Comment by adrianmroz** [comment]
Hi,

Unfortunately we introduced some regression in 1.9.0 for non trivial formulas for dimensions. It's fixed now in 1.9.1. If you get error on 1.9.0, please update. If it's on another version - give us the number.

**Comment by hollow** [comment]
Hi @adrianmroz,

I must have forked the repository a few minutes before the fixes for 1.9.1 have been pushed. It works as expected now, thank you so much 👍
